tests/testthat/test-expand.R

```diff
@@ -52,6 +52,20 @@ test_that("preserves NAs", {
   expect_equal(nesting(x)$x, x)
 })
 
+test_that("crossing preserves factor levels", {
```

Comment:
I think you should have one test for `crossing()`, but since the change is actually in `ulevel()` would you mind focussing most of your testing energy there?